torch/utils/cpp_extension.py

```diff
@@ -282,6 +282,28 @@ def _join_sycl_home(*paths) -> str:
     '-D__HIP_NO_HALF_CONVERSIONS__=1',
 ]
 
+_COMMON_SYCL_FLAGS = [
+    '-fsycl',
+    '-fsycl-targets=spir64_gen,spir64',
+]
+
+def _get_sycl_arch_list():
+    if 'TORCH_XPU_ARCH_LIST' in os.environ:
+        return os.environ.get('TORCH_XPU_ARCH_LIST')
+    arch_list = torch.xpu.get_arch_list()
+    # Dropping dg2-* archs since they lack hardware support for fp64 and require
+    # special consideration from the user. If needed these platforms can
+    # be requested thru TORCH_XPU_ARCH_LIST environment variable.
+    arch_list = [x for x in arch_list if not x.startswith('dg2-')]
+    return ','.join(arch_list)
+
+_SYCL_DLINK_FLAGS = [
+    *_COMMON_SYCL_FLAGS,
+    '-fsycl-link',
+    '--offload-compress',
+    f'-Xs "-device {_get_sycl_arch_list()}"',
```

Comment:
Two remarks: this is computed at import time, therefore, as most of the users will not have `TORCH_XPU_ARCH_LIST` calls `torch.xpu.get_arch_list()` which causes extension compilation to fail on platforms that compiled without Sycl

Reply:
`ciflow/trunk` label was missing on this PR before the merge was triggered. So, we've missed the failure. Thank you for the quick notice.